alma/common/install_nccl.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+set -ex
+
+# Install NCCL
+yum install -y rpm-build rpmdevtools
+NCCL_VERSION="2.14.3-1"
+TARBALL="v${NCCL_VERSION}.tar.gz"
+NCCL_DOWNLOAD_URL=https://github.com/NVIDIA/nccl/archive/refs/tags/${TARBALL}
+pushd /tmp
+wget ${NCCL_DOWNLOAD_URL}
+tar -xvf ${TARBALL}
+
+pushd nccl-${NCCL_VERSION}
+make -j src.build
+make pkg.redhat.build
+rpm -i ./build/pkg/rpm/x86_64/libnccl-${NCCL_VERSION}+cuda11.6.x86_64.rpm
+echo "exclude=libnccl" | tee -a /etc/yum.conf
+rpm -i ./build/pkg/rpm/x86_64/libnccl-devel-${NCCL_VERSION}+cuda11.6.x86_64.rpm
+echo "exclude=libnccl-devel" | tee -a /etc/yum.conf
+rpm -i ./build/pkg/rpm/x86_64/libnccl-static-${NCCL_VERSION}+cuda11.6.x86_64.rpm
+echo "exclude=libnccl-static" | tee -a /etc/yum.conf
+popd
+
+# Install the nccl rdma sharp plugin
+mkdir -p /usr/local/nccl-rdma-sharp-plugins
+git clone https://github.com/Mellanox/nccl-rdma-sharp-plugins.git
```

Comment:
Should we be pulling from a fixed release for these plugins just to keep track of that (and same question for the nccl test)?

Reply:
The NCCL RDMA sharp repository had a last release on Dec 27, 2020 but the master branch seems active with recent updates. Besides, it is a change across distros we support. I think it'll be a good idea to address it in the upcoming PR.